packages/strapi-design-system/src/TextButton/TextButton.tsx

```diff
@@ -66,27 +59,23 @@ export const TextButton = React.forwardRef<HTMLButtonElement, TextButtonProps>(
         onClick={handleClick}
         as="button"
         type="button"
+        background="transparent"
+        gap={2}
         {...props}
       >
-        {(startIcon || loading) && (
-          <Box as="span" paddingRight={2} aria-hidden>
-            {loading ? (
-              <LoadingWrapper>
-                <Loader />
-              </LoadingWrapper>
-            ) : (
-              startIcon
-            )}
-          </Box>
+        {loading ? (
+          <LoadingWrapper aria-hidden>
```

Comment:
Actually: I don't think this should have `aria-hidden`. Wdyt @joshuaellis ?

Reply:
Interesting question, is it just an icon? If so then I think it's fine, but if it supposed to tell people "this is loading" then probably not, but I don't know if adding to its label is correct either...

Reply:
I think if this component is in a loading state, which is only indicated by the icon, it shouldn't be hidden. On the other hand `Loader` defines `role="alert" aria-live="assertive"` - I could imagine these don't relate to whether the element is hidden or nor and fire anyway, which would then inform the user. However: if this is the case why define it as hidden at all?

Reply:
> I think if this component is in a loading state, which is only indicated by the icon, it shouldn't be hidden

But if it's just an SVG it doesn't mean anything to a screen-reader without a label you can see that [here](https://design-system-git-main-strapijs.vercel.app/?path=/story/design-system-components-textbutton--loading)

Also it only uses the icon `Loader` not the component so there's no aria role // announcement which i think is okay, in reality you'd expect the label for button to change, `save` -> `saving`. We could add `aria-busy`? but im not sure on that one either....

This issue looks like it could be useful to read through? https://github.com/Shopify/polaris/issues/1212

Reply:
Let's leave it as it is and create a follow-up for this. I could imagine this also applies to all other buttons and it might make sense to work out what we want globally. Would that work for you?

Reply:
Yeah i think is a topic that could benefit from some research 🔍